broker.go

```diff
@@ -157,23 +270,32 @@ type Pipeline struct {
 
 // RegisterPipeline adds a pipeline to the broker.
 func (b *Broker) RegisterPipeline(def Pipeline) error {
+	err := def.validate()
+	if err != nil {
+		return err
```

Comment:
do we want to test this error case?

Reply:
I believe `TestPipelineValidate` covers the validation function, I dunno if there's much extra to be gained in testing `RegisterPipeline` also, as it would just be testing that the `if` works (unless we wanted to assert that `validate()` was called? 🤔